crates/polars-io/src/ipc/ipc_stream.rs

```diff
@@ -293,18 +293,27 @@ impl<W: Write + Seek> IpcStreamBatchedWriter<W> {
     }
 
     /// WARNING: calling this if you opened the File without read permissions will hang!
-    pub fn finish(mut self, use_stream_position: bool) -> PolarsResult<usize> {
+    pub fn finish(mut self) -> PolarsResult<usize> {
+        self.writer
+            .finish()
+            .map(|continuation_size| self.total_written_bytes += continuation_size)?;
+
+        Ok(self.total_written_bytes)
+    }
+}
+
+impl<W: Write + Seek> IpcStreamBatchedWriter<W> {
+    pub fn finish_with_stream_position(mut self) -> PolarsResult<usize> {
         self.writer
             .finish()
             .map(|continuation_size| self.total_written_bytes += continuation_size)?;
 
-        if use_stream_position {
-            if let Ok(pos) = self.writer.into_inner().stream_position() {
-                return Ok(pos as usize);
-            } else {
-                eprintln!("Could not get stream position, defaulting to estimated size");
-            }
+        if let Ok(pos) = self.writer.into_inner().stream_position() {
+            return Ok(pos as usize);
+        } else {
+            eprintln!("Could not get stream position, defaulting to estimated size");
```

Comment:
is the estimated size guaranteed to be an over-estimation?

Reply:
Are you asking if it will be more?